File: C++ Programming/Competitive Programming/Coding Challenges 1/program2.cpp
/*
Q2.Accept Number of Rows and Number of columns from user and display below pattern.
Input : iRow = 4 iCol = 4
Output :* * * *
        * * * #
        * * # #
        * # # #
*/
#include<iostream>
using namespace std;

void Pattern(int iRow,int iCol)
{
    int i=4;
    int j=0;

    while ( i >= 1)
    {
        j=1;
        while (  j <= iCol )
        {
            if (j<=i)
            {
                cout<<"* ";
            }
            else
            {
                cout<<"# ";
            }
            j++;
        }
        cout<<"\n";
        i--;
    }
}
int main()
{
    int iValue1=0,iValue2=0;

    cout<<"Enter The Number of Row: ";
    cin>>iValue1;

    cout<<"Enter The Number of Column: ";
    cin>>iValue2;

    Pattern(iValue1,iValue2);

    return 0;
}

File: Java Programming/Competitive Programming/Coding Challenges 6/program2.java
import java.util.Scanner;

class program2 
 {
    public static void Pattern(int iRow, int iCol) 
    {
        int i=1;
        int j=0;
        do {
            j=1;
            do {
                if (j <= iCol - i +1) 
                {
                    System.out.print("* ");
                }
                else
                {
                    System.out.print("# ");
                }
                j++;
            } 
            while (j<=iCol);
            System.out.println();
        i++; 
        } 
        while (i<=iRow);  
    }
    public static void main(String[] args) 
    {
        Scanner sobj=new Scanner(System.in);
        int iValue1=0,iValue2=0;

        System.out.println("Enter The Number of Row: ");
        iValue1=sobj.nextInt();

        System.out.println("Enter The Number of Column: ");
        iValue2=sobj.nextInt();

        Pattern(iValue1,iValue2);

        sobj.close();
        
    }
}
